sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI
+{
+    /// <summary>
+    /// A representation of a full response to a chat completions request.
+    /// </summary>
+    /// <remarks>
+    /// Chat completions can contain one or more chat choices, each holding a message with string content and an
+    /// associated role. When requesting more than one completion, a corresponding number of additional choices will
+    /// appear in the chat completions result.
+    /// </remarks>
+    public partial class ChatCompletions
```

Comment:
I know what you mean!
Strictly/technically, `/completions` and `/chat/completions` are entirely distinct REST endpoints and it's a "coincidence" that they look so similar; at least academically, nothing stops a breaking change from causing chat to look entirely distinct from pre-chat completions -- at which point we'd be in a pickle if we used inheritance. Practically, that's very unlikely to ever be a thing, but it may influence the choice (I think it's still subject to plenty of discussion).

Language-wise, the trouble is that we'd need to abstract the "prompt" payload since we can't hide/replace a member from the base class; that's very doable to design, but it starts getting into territory where we might be taking too many liberties for a lower-level client. But I'm not sure--there's also a lot of pain in duplicating so much code.

Reply:
This seems to have become even truer after looking more closely and seeing that several fields (echo, best_of, logprobs) were removed. I'll still like to see if there's a way we could consolidate objects without the SDK trying to be too smart for its own good, but that's likely a follow-up question.